packages/core/src/components/carousel/carousel.tsx

```diff
@@ -1,142 +1,198 @@
+    this.nextButton = this.element.shadowRoot.querySelector('.navigation--next')
+    this.prevButton = this.element.shadowRoot.querySelector('.navigation--prev')
+
+    this.carouselWrapper.addEventListener('touchstart', this.handleTouchStart)
+    this.carouselWrapper.addEventListener('touchend', this.handleTouchEnd)
+
+    this.showOrHideNavigationButtons()
+
+    if (this.autoplay > 0) this.startAutoplay()
+  }
+
+  disconnectedCallback() {
+    this.carouselWrapper.removeEventListener(
+      'touchstart',
+      this.handleTouchStart
+    )
+    this.carouselWrapper.removeEventListener('touchend', this.handleTouchEnd)
+  }
 
-    if (this.paginationType === 'thumbnails') {
-      const urls = this.thumbnailImages ? this.thumbnailImages.split(',') : []
+  handleMoveToPaginationItem(index: number) {
+    if (!this.carouselItems || !this.carouselItems[index]) return
 
-      params = {
-        pagination: {
-          ...params.pagination,
-          renderCustom: (_swiper, current, total) =>
-            renderThumbs(current, total, urls, this.videoIcons),
-        },
-      }
+    this.currentIndex = index
+    this.showOrHideNavigationButtons()
+  }
+
+  handleNavigationClick(event: Event) {
+    const button = event.currentTarget as HTMLButtonElement
+    const isNext = button?.classList.contains('navigation--next')
+    let newIndex = this.currentIndex + (isNext ? 1 : -1)
+
+    const isLastItemAndNext =
+      isNext && this.currentIndex === this.carouselItems.length - 1
+    const isFirstItemAndPrev = !isNext && this.currentIndex === 0
+
+    if (!this.loop && (isLastItemAndNext || isFirstItemAndPrev)) return
+
+    if (this.loop) {
+      newIndex =
+        (newIndex + this.carouselItems.length) % this.carouselItems.length
     }
 
-    Object.assign(this.swiperEl, params)
+    this.currentIndex = newIndex
+    this.showOrHideNavigationButtons()
   }
-  render() {
-    const injectStyles = [
-      '.swiper-button-disabled { opacity: 0 !important}',
-      '.swiper-pagination-custom { display: flex; justify-content: center; column-gap: var(--spacing-base); } ',
-      `.swiper-button-next { width: var(--spacing-${this.navigationButtonSize}); height: var(--spacing-${this.navigationButtonSize}); }`,
-      `.swiper-button-prev { width: var(--spacing-${this.navigationButtonSize}); height: var(--spacing-${this.navigationButtonSize}); }`,
-    ]
 
+  showOrHideNavigationButtons() {
+    const hasButtons = this.nextButton && this.prevButton
+
+    if (!hasButtons || !this.hasNavigation || this.loop) return
+
+    this.nextButton.classList.toggle(
+      'disabled',
+      this.currentIndex === this.carouselItems.length - 1
+    )
+    this.prevButton.classList.toggle('disabled', this.currentIndex === 0)
+  }
+
+  startAutoplay() {
+    this.loop = true
+
+    setInterval(() => {
+      this.currentIndex = (this.currentIndex + 1) % this.carouselItems.length
+      this.showOrHideNavigationButtons()
+    }, this.autoplay)
+  }
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏We can store the loop interval in a variable so we can clear it and prevent memory leaks, in case there are changes in the autoplay. But I imagine this will only happen when people change the autoplay attribute a lot. nevermind I noticed that we only pass it when the component initializes, but we need to clear it when it unmounts.